controllers/pulp/ingress.go

```diff
@@ -0,0 +1,230 @@
+	json.Unmarshal([]byte(cmdOutput), &pulpPlugins)
+	defaultPlugins := []IngressPlugin{
+		{
+			Name:        pulp.Name + "-content",
+			Path:        getPulpSetting(pulp, "content_path_prefix"),
+			TargetPort:  "content-24816",
+			ServiceName: pulp.Name + "-content-svc",
+		},
+		{
+			Name:        pulp.Name + "-api-v3",
+			Path:        getPulpSetting(pulp, "api_root") + "api/v3/",
+			TargetPort:  "api-24817",
+			ServiceName: pulp.Name + "-api-svc",
+		},
+		{
+			Name:        pulp.Name + "-auth",
+			Path:        "/auth/login/",
+			TargetPort:  "api-24817",
+			ServiceName: pulp.Name + "-api-svc",
+		},
+		{
+			Name:        pulp.Name,
+			Path:        "/",
+			TargetPort:  "api-24817",
+			ServiceName: pulp.Name + "-api-svc",
+		},
+	}
+	pulpPlugins = append(defaultPlugins, pulpPlugins...)
+
+	// get ingress
+	pulpIngress := &netv1.Ingress{}
+	err = r.Get(ctx, types.NamespacedName{Name: pulp.Name, Namespace: pulp.Namespace}, pulpIngress)
+
+	// Create the ingress in case it is not found
+	if err != nil && errors.IsNotFound(err) {
+		ingressObj := pulpIngressObject(pulp, pulpPlugins)
+		ctrl.SetControllerReference(pulp, ingressObj, r.Scheme)
+		log.Info("Creating a new ingress", "Ingress.Namespace", ingressObj.Namespace, "Ingress.Name", ingressObj.Name)
+		r.updateStatus(ctx, pulp, metav1.ConditionFalse, pulp.Spec.DeploymentType+"-Ingress-Ready", "CreatingIngress", "Creating "+pulp.Name+"-ingress")
+		err = r.Create(ctx, ingressObj)
+		if err != nil {
+			log.Error(err, "Failed to create new ingress", "Ingress.Namespace", ingressObj.Namespace, "Ingress.Name", ingressObj.Name)
+			r.updateStatus(ctx, pulp, metav1.ConditionFalse, pulp.Spec.DeploymentType+"-Ingress-Ready", "ErrorCreatingIngress", "Failed to create "+pulp.Name+"-ingress: "+err.Error())
+			r.recorder.Event(pulp, corev1.EventTypeWarning, "Failed", "Failed to create new ingress")
+			return ctrl.Result{}, err
+		}
+	} else if err != nil {
+		log.Error(err, "Failed to get ingress")
+		return ctrl.Result{}, err
+	}
+
+	// we should only update the status when Ingress-Ready==false
+	if v1.IsStatusConditionFalse(pulp.Status.Conditions, cases.Title(language.English, cases.Compact).String(pulp.Spec.DeploymentType)+"-Ingress-Ready") {
+		r.updateStatus(ctx, pulp, metav1.ConditionTrue, pulp.Spec.DeploymentType+"-Ingress-Ready", "IngressTasksFinished", "All Ingress tasks ran successfully")
+		r.recorder.Event(pulp, corev1.EventTypeNormal, "IngressReady", "All Ingress tasks ran successfully")
+	}
+	return ctrl.Result{}, nil
+}
+
+// pulp-ingress
+func pulpIngressObject(m *repomanagerv1alpha1.Pulp, plugins []IngressPlugin) *netv1.Ingress {
+	annotation := map[string]string{
+		"kubernetes.io/ingress.class":         m.Spec.IngressClass,
+		"haproxy.router.openshift.io/timeout": m.Spec.HAProxyTimeout,
+	}
+	var paths []netv1.HTTPIngressPath
+	var path netv1.HTTPIngressPath
+	pathType := netv1.PathTypePrefix
+	rewrite := ""
+	if strings.ToLower(m.Spec.IngressClass) == "nginx" {
```

Comment:
What if we create a function to check ingress controller type based on the IngressClass `.spec.controller == k8s.io/ingress-nginx`?

This could avoid the issue with different class names for the same controller type (nginx) and could be reused in other places like:
https://github.com/pulp/pulp-operator/blob/9dfd086ede337953eabaf8b08be9be368f2e2dfc/controllers/pulp/ingress.go#L155
https://github.com/pulp/pulp-operator/blob/9dfd086ede337953eabaf8b08be9be368f2e2dfc/controllers/pulp/status.go#L83
https://github.com/pulp/pulp-operator/blob/9dfd086ede337953eabaf8b08be9be368f2e2dfc/controllers/pulp/ingress.go#L155

Reply:
sounds good to me, but where can I get the controller data?